doc/api/animation_api.rst

```diff
@@ -211,16 +211,16 @@ debug.
    FFMpegFileWriter
    ImageMagickFileWriter
 
-Fundamentally, a `MovieWriter` provides a way to grab sequential frames
-from the same underlying `~matplotlib.figure.Figure` object.  The base
-class `MovieWriter` implements 3 methods and a context manager.  The
-only difference between the pipe-based and file-based writers is in the
+Fundamentally, an `AbstractMovieWriter` provides a way to grab sequential
+frames from the same underlying `~matplotlib.figure.Figure` object.  The
+base `AbstractMovieWriter` class implements 3 methods and a context manager.
+The only difference between the pipe-based and file-based writers is in the
 arguments to their respective ``setup`` methods.
```

Comment:
```suggestion
The writer classes provide a way to grab sequential frames from the same
underlying `~matplotlib.figure.Figure` object.  They all provide 3 methods
that must be called in sequence:

- `~-AbstractMovieWriter.setup` prepares the writer (e.g. opening a pipe).
  Pipe-based and file-based writers take different arguments to ``setup()``.
- `~.AbstractMovieWriter.grab_frame` can then be called as often as
  needed to capture a single frame at a time
- `~.AbstractMovieWriter.finish` finalizes the movie and writes the output
  file to disk.
```

Reply:
IMHO it doesn't make sense to specifically mention `AbstractMovieWriter` because nobody will use that class. Instead, we can more informally say "all writer classes".

Note also that #21954 moves the ineritance diagram here and adds missing classes, which should make the relations more clear.